pkg/autoscaler/metrics/http_scrape_client_test.go

```diff
@@ -250,3 +268,37 @@ func newTestHTTPClient(response *http.Response, err error) *http.Client {
 		},
 	}
 }
+
+func BenchmarkUnmarshallingProtoData(b *testing.B) {
+	benchmarks := []struct {
+		name string
+		resp *http.Response
+	}{{
+		name: "BenchmarkStatWithEmptyPodName",
+		resp: makeProtoResponseForBenchmarking(http.StatusOK, "", network.ProtoAcceptContent),
```

Comment:
You can drop the entire helper function now and just throw a `Stat` in here that you'll marshall right before the test and throw in a buffer like you did I think.